oauth2client/client.py

```diff
@@ -1222,6 +1229,31 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        # This import needs to be here to avoid a circular dependency.
+        from oauth2client.service_account import _ServiceAccountCredentials
+        data = json.loads(_from_bytes(s))
+
+        # We handle service_account._ServiceAccountCredentials since it is a
+        # possible return type of GoogleCredentials.get_application_default()
+        if (data['_module'] == 'oauth2client.service_account' and
+            data['_class'] == '_ServiceAccountCredentials'):
+            return _ServiceAccountCredentials.from_json(s)
```

Comment:
@nathanielmanistaatgoogle You're picking the wrong time to stick. The right time would've been when [`GoogleCredentials.get_application_default`][1] was added.

It
- [can return][2] an [`oauth2client.contrib.gce.AppAssertionCredentials`][4]
- [can return][3] an [`oauth2client.contrib.appengine.AppAssertionCredentials`][5]
- a [`GoogleCredentials`][6]
- an [`oauth2client.service_account._ServiceAccountCredentials`][7]

[1]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L1340
[2]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L1253
[3]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L1236
[4]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/contrib/gce.py#L39
[5]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/contrib/appengine.py#L141
[6]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L1484
[7]: https://github.com/google/oauth2client/blob/8f8df1f5977ec76d50cb3dca9a31044cfd1fb9b4/oauth2client/client.py#L1493